File: java/src/main/java/com/fengtian/qianer/atcoder/abc/abc163/ABC163_C.java
package com.fengtian.qianer.atcoder.abc.abc163;

import java.util.Arrays;
import java.util.Scanner;

public class ABC163_C {

    public static void main(String[] args) {
        final Scanner sc = new Scanner(System.in);

        final int n = sc.nextInt();

        final int[] ans = new int[n];
        for (int i = 2; i <= n; i++) {
            final int a = sc.nextInt();
            ans[a-1]++;
        }

        Arrays.stream(ans).forEach(System.out::println);
    }
}

File: kotlin/src/abc/abc163/ABC163_C.kt
package abc.abc163

import java.util.*

fun main(args: Array<String>) {
    val sc = Scanner(System.`in`)

    val n = sc.nextInt()

    val map = mutableMapOf<Int, Int>()

    (1..n).forEach { map.computeIfAbsent(it) { 0 } }
    (2..n).forEach { map.computeIfPresent(sc.nextInt()) { k, v -> v+1 } }

    map.values.forEach(::println)
}